lib/sqlitex/query.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Sqlitex.Query do
+  def bind_values({:error, _}=error, _opts), do: error
+  def bind_values({:ok, statement}, opts) do
+    params = params_from_opts(opts)
+    case :esqlite3.bind(statement, params) do
+      {:error,_}=error -> error
+      :ok -> {:ok, statement}
+    end
+  end
+
+  def execute({:error,_}=error, _opts), do: {error, {}, {}, []}
+  def execute({:ok, statement}, opts) do
+    into = into_from_opts(opts)
+    types = :esqlite3.column_types(statement)
+    columns = :esqlite3.column_names(statement)
+    data = :esqlite3.fetchall(statement)
+    {types, columns, data, into}
+  end
+
+  def prepare(sql, database), do: :esqlite3.prepare(sql, database)
```

Comment:
The pipe changes you are doing might make this obsolete, but you could change `prepare/2` to:

```elixir
def prepare(sql, database) do
  {:ok, stmt} = :esqlite3.prepare(sql, database)
  stmt
end
```

And now you don't have to worry about catching the `:ok` value as an argument to `bind_values/2`.